### DISCUSSION TITLE
Bug

### DISCUSSION THREAD

**Discussion by GalonMitsunari** [codefori/vscode-ibmi#1856] [Q&A]
I've been using the extension to my business as a junior developer for a few months now.
Since yesterday, I've been able to connect to the extension but I'm unable to see my libraries and files. When I connect, on the left of VScode there's a pop-up with a red cross and an error 42601.
![image_2024-02-20_081013412](https://github.com/codefori/vscode-ibmi/assets/56174754/5764d33f-2e59-455f-9cef-47ec4e742f46)
![image_2024-02-20_081023577](https://github.com/codefori/vscode-ibmi/assets/56174754/d1e2b136-030d-4201-a3da-8e6b2583c8bc)

**Comment by sebjulliand** [comment]
Hi @GalonMitsunari ,
please share the content of your Code for IBM i output after the error shows up:
![code for i output](https://github.com/codefori/vscode-ibmi/assets/11096890/88bc7363-2402-419f-b173-309358b914b9)

Also, checkout the Console in the developer tools and send any error that would look related:
![image](https://github.com/codefori/vscode-ibmi/assets/11096890/a417a06f-4aab-451b-a5fa-6875856e74d4)

![image](https://github.com/codefori/vscode-ibmi/assets/11096890/6ebc6e07-d4af-47a8-ab6d-903bc2701e2d)

Finally, what is your IBM i OS version?

**Reply by GalonMitsunari** [reply]
Here are my screenshots of my errors, I don't know my ibm version.
![image_2024-02-20_102547854](https://github.com/codefori/vscode-ibmi/assets/56174754/20dcf01e-d20e-48cb-b682-714f69828f27)
![image_2024-02-20_102559542](https://github.com/codefori/vscode-ibmi/assets/56174754/25939bd3-5caa-46f7-9fc5-feecf089e201)
![image_2024-02-20_102625217](https://github.com/codefori/vscode-ibmi/assets/56174754/f0349e44-3a50-42b0-8071-643f9126f99f)
![image_2024-02-20_102642038](https://github.com/codefori/vscode-ibmi/assets/56174754/06cd2f5b-1257-4437-a9f7-915209c99979)
![image_2024-02-20_102717891](https://github.com/codefori/vscode-ibmi/assets/56174754/36fb93ee-5264-4802-be95-fe4688c8580d)
![image_2024-02-20_102733358](https://github.com/codefori/vscode-ibmi/assets/56174754/0f5b3037-44be-409c-9567-98547a0be2b5)
![image_2024-02-20_102911969](https://github.com/codefori/vscode-ibmi/assets/56174754/6aebf9b7-c5c4-40df-83b8-499ee8fe9019)

**Reply by sebjulliand** [reply]
Thank you. You seem to be running on an old version of the OS. Anything below 7.3 will not work with Code for IBM i.
Can you get that OS version using this command and put the result here? Thanks.

`DSPDTAARA QUSRSYS/QSS1MRI`

**Comment by GalonMitsunari** [comment]
it returns this: V7R1M000 2928
however, before yesterday and on other days, the extension worked, and it works on other as/400s

**Reply by sebjulliand** [reply]
That probably used to work before release `2.7.0` went out. Some changes introduced in it are not compatible with 7.1 at all. Your other IBM i are probably at level 7.2 or above, I assume.
If getting this IBM i eventually updated (which would be for the best 😅) is not an option, then re-installing the version of the extension that worked for you is your best bet.
![image](https://github.com/codefori/vscode-ibmi/assets/11096890/145f3f55-db7e-4cce-901c-ada9e482c407)

**Reply by GalonMitsunari** [reply]
I've just changed versions and I'm still getting the same errors.
